File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        int a = 0, b =  0;
        // [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
        System.out.println("Hello Java");
        
        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == '('){
                a++;
            } else {
                a--;
            }
            if(a < 0){
                answer = false;
            }
        }
        if(a != 0){
                answer = false;
            }

        return answer;
    }
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    
    # [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    ans_list = []
    for ch in s:
        if ch == '(':
            ans_list.append(ch)
        else:
            try:
                ans_list.pop()
            except:
                return False
    
    if len(ans_list) == 0:
        return True
    else:
        return False